Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -202,34 +202,54 @@ extension NavigationMapView {
         if traveledDifference == 0.0 {
             return
         }
-        let startDate = Date()
-        vanishingRouteLineUpdateTimer = Timer.scheduledTimer(withTimeInterval: 0.05, repeats: true, block: { timer in
-            let timePassedInMilliseconds = Date().timeIntervalSince(startDate) * 1000
-            if timePassedInMilliseconds >= 980 {
-                timer.invalidate()
-                return
-            }
-            
-            let newFractionTraveled = self.preFractionTraveled + traveledDifference * timePassedInMilliseconds.truncatingRemainder(dividingBy: 1000) / 1000
-            
+        switch puckType {
+        case .default:
+            let startDate = Date()
+            vanishingRouteLineUpdateTimer = Timer.scheduledTimer(withTimeInterval: 0.05, repeats: true, block: { timer in
+                let timePassedInMilliseconds = Date().timeIntervalSince(startDate) * 1000
+                if timePassedInMilliseconds >= 980 {
+                    timer.invalidate()
+                    return
+                }
+                
+                let newFractionTraveled = self.preFractionTraveled + traveledDifference * timePassedInMilliseconds.truncatingRemainder(dividingBy: 1000) / 1000
+                
+                do {
+                    try self.mapView.mapboxMap.style.updateLayer(withId: mainRouteLayerIdentifier) { (lineLayer: inout LineLayer) throws in
+                        let congestionSegments = routeProgress.route.congestionFeatures(legIndex: self.currentLegIndex, roadClassesWithOverriddenCongestionLevels: self.roadClassesWithOverriddenCongestionLevels)
+                        let mainRouteLayerGradient = self.routeLineGradient(congestionSegments,
+                                                                            fractionTraveled: newFractionTraveled)
+                        
+                        lineLayer.lineGradient = .expression(Expression.routeLineGradientExpression(mainRouteLayerGradient))
+                    }
+                    
+                    try self.mapView.mapboxMap.style.updateLayer(withId: mainRouteCasingLayerIdentifier) { (lineLayer: inout LineLayer) throws in
+                        let mainRouteCasingLayerGradient = self.routeLineGradient(fractionTraveled: newFractionTraveled)
+                        
+                        lineLayer.lineGradient = .expression(Expression.routeLineGradientExpression(mainRouteCasingLayerGradient))
+                    }
+                } catch {
+                    NSLog("Failed to update main route line layer.")
```

Comment:
Should we swallow the error like this, or does an error updating the layer indicate a more serious internal inconsistency in the SDK?

If we should log, use `print(_:)` instead of `NSLog(_:)`.

Reply:
thanks for the suggestion, I'll update the log to `print`